sources/ElkArte/Graphics/Imagick.php

```diff
@@ -0,0 +1,398 @@
+<?php
+
+/**
+ * This file deals with low-level graphics operations performed on images,
+ * specially as needed for the Imagick library
+ *
+ * @package   ElkArte Forum
+ * @copyright ElkArte Forum contributors
+ * @license   BSD http://opensource.org/licenses/BSD-3-Clause (see accompanying LICENSE.txt file)
+ *
+ * @version 2.0 dev
+ *
+ */
+
+namespace ElkArte\Graphics;
+
+class Imagick extends AbstractManipulator
+{
+	/** @var \Imagick */
+	protected $_image;
+
+	public function __construct($image)
+	{
+		$this->setSource($image);
+
+		try
+		{
+			$this->memoryCheck();
+		}
+		catch (\Exception $e)
+		{
+			return false;
+		}
+
+		return true;
+	}
+
+	public function setSource($source)
+	{
+		$this->_fileName = $source;
+	}
+
+	/**
+	 * Checks whether the Imagick class is present.
+	 *
+	 * @return bool Whether or not the Imagick extension is available.
+	 */
+	public static function canUse()
+	{
+		return class_exists('\Imagick', false);
+	}
+
+	public function __destruct()
+	{
+		if ($this->_image)
+		{
+			$this->_image->clear();
+		}
+	}
+
+	/**
+	 * Loads a image file into the image engine for processing
+	 *
+	 * @return bool|mixed
+	 */
+	public function createImageFromFile()
+	{
+		$this->getSize();
+
+		if (isset(Image::DEFAULT_FORMATS[$this->sizes[2]]))
+		{
+			try
+			{
+				$this->_image = new \Imagick($this->_fileName);
+			}
+			catch (\Exception $e)
+			{
+				return false;
+			}
+		}
+		else
+		{
+			return false;
+		}
+
+		$this->_setImage();
+
+		return true;
+	}
+
+	/**
+	 * Sets the image sizes.
+	 */
+	protected function _setImage()
+	{
+		// Update the image size values
+		$this->_width = $this->sizes[0] = $this->_image->getImageWidth();
+		$this->_height = $this->sizes[1] = $this->_image->getImageHeight();
+	}
+
+	/**
+	 * Loads an image from a web address into the image engine for processing
+	 *
+	 * @return bool|mixed
+	 */
+	public function createImageFromWeb()
+	{
+		require_once(SUBSDIR . '/Package.subs.php');
+		$this->_image = fetch_web_data($this->_fileName);
```

Comment:
Shouldn't be $blob = fetch_?

Reply:
As above ... its a result of $this->getSize('string'); wanting $this->_image to be populated.  And as above to fix would need to pass the value instead.